File: src/services/api/http/seller.ts
import type { ISellerApi, SellerStats, SellerDiscordSettings } from "@/services/api.types";
import type { PaginatedResponse, SellerPlan, SellerMember, CrosscheckRow, TimelineEvent, PlatformWebhookEvent, DiscordValidationResult, SystemAnnouncement } from "@/types";
import { httpClient } from "./client";

export const sellerApi: ISellerApi = {
    getStats: () => httpClient.get<SellerStats>("/api/seller/stats"),
    getAnnouncements: () => httpClient.get<SystemAnnouncement[]>("/api/seller/announcements"),
    getDiscordSettings: () => httpClient.get<SellerDiscordSettings>("/api/seller/discord/settings"),
    getPlans: (params?) => httpClient.get<SellerPlan[]>("/api/seller/plans", params as Record<string, string | number | boolean>),
    getPlanById: (id) => httpClient.get<SellerPlan | null>(`/api/seller/plans/${id}`),
    savePlan: (data) => httpClient.post<SellerPlan>("/api/seller/plans", data),
    getMembers: (params) => httpClient.get<PaginatedResponse<SellerMember>>("/api/seller/members", params as Record<string, string | number | boolean>),
    getMemberById: (id) => httpClient.get<SellerMember | null>(`/api/seller/members/${id}`),
    getMemberTimeline: (id) => httpClient.get<TimelineEvent[]>(`/api/seller/members/${id}/timeline`),
    getCrosscheck: (params?) => httpClient.get<CrosscheckRow[]>("/api/seller/crosscheck", params as Record<string, string | number | boolean>),
    runCrosscheck: () => httpClient.post<{ jobId: string }>("/api/seller/crosscheck/run"),
    getWebhooks: (params) => httpClient.get<PaginatedResponse<PlatformWebhookEvent>>("/api/seller/webhooks", params as Record<string, string | number | boolean>),
    validateDiscord: (guildId, roleId) => httpClient.post<DiscordValidationResult>("/api/seller/discord/validate", { guildId, roleId }),
};

File: backend/app/api/endpoints/seller.py
from fastapi import APIRouter, Query
from typing import Any
from app.schemas.common import PaginatedResponse
from pydantic import BaseModel

router = APIRouter()

class DiscordValidateRequest(BaseModel):
    guildId: str
    roleId: str | None = None

@router.get("/stats")
async def get_stats() -> Any:
    return {
        "totalMembers": 500,
        "activePlans": 3,
        "mrr": 200000,
        "churnRate": 0.05,
        "newMembersThisMonth": 20,
        "webhooksToday": 15
    }

@router.get("/plans")
async def get_plans() -> list[Any]:
    return [
        {"id": "p1", "name": "Basic Plan", "price": 1000, "status": "active", "stripeProductId": "prod_1", "roleId": "r1"}
    ]

@router.post("/plans")
async def create_plan(plan: dict) -> Any:
    return {"id": "p_new", "status": "success", **plan}

@router.get("/plans/{plan_id}")
async def get_plan_by_id(plan_id: str) -> Any:
    return {"id": plan_id, "name": "Basic Plan", "price": 1000, "status": "active", "stripeProductId": "prod_1", "roleId": "r1"}

@router.put("/plans/{plan_id}")
async def update_plan(plan_id: str, plan: dict) -> Any:
    return {"id": plan_id, "status": "success", **plan}

@router.get("/members")
async def get_members(page: int = Query(1, ge=1), page_size: int = Query(10, ge=1, le=100)) -> PaginatedResponse[Any]:
    return PaginatedResponse(
        items=[{"id": "m1", "name": "Member 1", "email": "m1@example.com", "planId": "p1", "billingStatus": "active", "discordUsername": "member1#1234", "discordRoleGranted": True, "joinedAt": "2024-01-10"}],
        page=page,
        page_size=page_size,
        total_count=1
    )

@router.get("/members/{member_id}")
async def get_member_by_id(member_id: str) -> Any:
    return {"id": member_id, "name": "Member 1", "email": "m1@example.com"}

@router.get("/members/{member_id}/timeline")
async def get_member_timeline(member_id: str) -> list[Any]:
    return []

@router.get("/crosscheck")
async def get_crosscheck() -> list[Any]:
    return []

@router.post("/crosscheck/sync")
async def sync_crosscheck() -> Any:
    return {"status": "success", "syncedCount": 5}

@router.get("/webhooks")
async def get_webhooks(page: int = Query(1, ge=1), page_size: int = Query(10, ge=1, le=100)) -> PaginatedResponse[Any]:
    return PaginatedResponse(items=[], page=page, page_size=page_size, total_count=0)

@router.post("/discord/validate")
async def validate_discord(payload: DiscordValidateRequest) -> Any:
    return {
        "botInstalled": True,
        "manageRolesPermission": True,
        "roleExists": True if payload.roleId else False,
        "botRoleHierarchy": True if payload.roleId else False,
        "errorCode": None,
        "errorMessage": None
    }
